frame/staking/src/pallet/impls.rs

```diff
@@ -159,29 +179,28 @@ impl<T: Config> Pallet<T> {
 		})?;
 		let mut ledger = <Ledger<T>>::get(&controller).ok_or(Error::<T>::NotController)?;
 
+		// clean up older claimed rewards
 		ledger
-			.claimed_rewards
+			.legacy_claimed_rewards
 			.retain(|&x| x >= current_era.saturating_sub(history_depth));
+		<Ledger<T>>::insert(&controller, &ledger);
```

Comment:
In the `TargetList` line of work, we are creating a VERY IMPORTANT assumption that any time a `ledger`'s stake fields change, we call into `update_ledger`.

We should first fix this by making the `Ledger` have a private fileds, and only specific `update` methods that hold these assumptions.

This code is fone though, because we are only touching `legacy_claimed_rewards`.

If you follow the code, you will see that in `make_payout`, we call `make_payout`, which is right, but is a thin thread, if you know what I mean.

Reply:
Also somewhat supports the idea that claimed_rewards does not belong to the ledger.